app/models/thread.php

```diff
@@ -1,89 +1,132 @@
 <?php
-	class Thread extends AppModel
-	{
-		public $validation = array(
-			'title' => array(
-				'length' => array(
-					'validate_between', 1,30,
-				),
-			),
-		);
-
-		public static function getAll()
-		{
-			$threads = array();
-			$db = DB::conn();
-			$rows = $db->rows('SELECT * FROM thread');
-			
-			foreach ($rows as $row) {
-				$threads[] = new Thread($row);
-			}
-		
-			return $threads;
-		}
-
-		public static function get($id)
-		{
-			$db = DB::conn();
-
-			$row = $db->row('SELECT * FROM thread WHERE id = ?', array($id));
-
-			if (!$row) {
-				throw new RecordNotFoundException('no record found');
-			}
-
-			return new self($row);
-		}
-
-		public function getComments()
-		{
-			$comments = array();
-
-			$db= DB::conn();
-
-			$rows = $db->rows('SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC', array($this->id));
-
-			foreach ($rows as $row) {
-				$comments[] = new Comment($row);
-			}
-			return $comments;
-		}
-
-		public function write(Comment $comment) //Will enable us to insert data into the comment table
-		{
-			if (!$comment->validate()) {
-				throw new ValidationException('invalid comment');
-			}
-
-			$db = DB::conn();
-			$db->query('INSERT INTO comment SET thread_id = ?, username = ?, body = ?, created = NOW()',
-				array($this->id, $comment->username, $comment->body)
-			);
-		}
-
-		public function create(Comment $comment)
-		{
-			$this->validate();
-			$comment->validate();
-			if ($this->hasError() || $comment->hasError()){
-				throw new ValidationException('invalid thread or comment');
-			}
-			$db = DB::conn();
-			$db->begin();
-
-			//$params = array(
-			//	'title' => $this->title,
-			//);
-			//$db->insert('thread', $params);
-			//or
-			$db->query('INSERT INTO thread SET title = ?, created = NOW()', array($this->title));
-
-			$this->id = $db->lastInsertId(); //returns the latest inserted id within the function
-
-			//write first comment at the same time
-			$this->write($comment);
-
-			$db->commit();
-		}
-	}
-?>
\ No newline at end of file
+
+class Thread extends AppModel
+{
+    const MIN_LENGTH = 1;
+    const MAX_LENGTH = 50;
+
+    public $validation = array(
+        'title' => array(
+            'length' => array(
+                'validate_between', self::MIN_LENGTH, self::MAX_LENGTH,
+            ),
+        ),
+    );
+
+    public static function getAll()
+    {
+        $threads = array();
+        $db = DB::conn();
+        $rows = $db->rows('SELECT t.id, t.title, t.created, u.username, t.user_id FROM thread t 
+            INNER JOIN user u ON t.user_id=u.id 
+            ORDER BY t.latest DESC');
+            
+        foreach ($rows as $row) {
+            $threads[] = new Thread($row);
+        }
+    
+        return $threads;
+    }
+
+    public static function getMyThreads()
```

Comment:
What's the difference of getAll() from getMyThreads() ? Please clarify the function name po and remove redundancy by not adding 'Threads' as the name of the class is already Thread :bow: